Lib/importlib/_bootstrap_external.py

```diff
@@ -1635,3 +1635,29 @@ def _install(_bootstrap_module):
     supported_loaders = _get_supported_file_loaders()
     sys.path_hooks.extend([FileFinder.path_hook(*supported_loaders)])
     sys.meta_path.append(PathFinder)
+
+
+def alternative_architectures(suffixes):
+    """Add a suffix with an alternative architecture name
+    to the list of suffixes so an extension built with
+    the default (upstream) setting is loadable with our Pythons
+    """
+
+    arch_map = {
```

Comment:
Would it be slightly faster\* to have this as module level global constant instead (e.g. `_ARCH_MAP`)?

\* speaking about speed, how does this impact the general speed of imports? I'd be worried here.

Reply:
This module is baked into C header file during build so it should be fast enough. Also, finders and loaders for standard paths are initialized during a startup so it should not have any impact on `import`.

The worst-case scenario is for architectures without alternatives because it loops over all combinations but it means usually only 3 (suffixes) × 8 (alternatives in dict) = 24 loops. It might be possible to return from the function immediately for `x86_64` and `i686` architectures if you think it makes any difference.

Reply:
> This module is baked into C header file during build 

I don't think it is compiled to anything faster than Python bytecode when baking it in, correct?

> The worst-case scenario is for architectures without alternatives because it loops over all combinations but it means usually only 3 (suffixes) × 8 (alternatives in dict) = 24 loops

And this happens only once per interpreter startup, correct? Because the function is called twice in this file.

Reply:
> > This module is baked into C header file during build
> 
> I don't think it is compiled to anything faster than Python bytecode when baking it in, correct?

It uses marshal, so yes, byte-code.

> > The worst-case scenario is for architectures without alternatives because it loops over all combinations but it means usually only 3 (suffixes) × 8 (alternatives in dict) = 24 loops
> 
> And this happens only once per interpreter startup, correct? Because the function is called twice in this file.

The function is called twice during the startup.